plugins/content_distribution/batch/Distribute/engines/SftpDistributionEngine.php

```diff
@@ -30,7 +30,7 @@ protected function getTempDirectoryForProfile($distributionProfileId)
 	{
 		$tempFilePath = $this->tempDirectory . '/' . $this->getTempDirectory() . '/' . $distributionProfileId . '/';
 		if (!file_exists($tempFilePath))		
-			mkdir($tempFilePath, 0777, true);
+			kFile::fullMkfileDir($tempFilePath);
```

Comment:
I reverted this change last weeek since it caused issues. Please note that the function you are calling requires the full file path to create the directory for. In your case you are passing only the directory path.

Reply:
kFile::fullMkfileDir expects to get directory as input.